### PR TITLE
Highlight Missing dependencies and display object kind

### PR DESCRIPTION
**Pull request by aminelouati**
In this PR, we:
1. Highlight Missing dependencies
2. Display object kind within each vertex of the Call Graph

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java [hunk 1]
```diff
@@ -299,20 +320,33 @@ private void callGraphStyle(JGraphXAdapter<GraphNode, DefaultEdge> callGraphAdap
...
                 for (String dependsOnCatalogObject : dependsOnCatalogObjects) {
                     String bucketName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(0);
-                    String workflowName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(1);
-                    GraphNode calledWorkflow = callGraphHolder.addNode(bucketName, workflowName);
-                    callGraphHolder.addDependsOnEdge(callingWorkflow, calledWorkflow);
+                    String objectName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(1);
+                    boolean isCatalogObjectExist = catalogObjectService.isDependsOnObjectExistInCatalog(bucketName,
+                                                                                                        objectName,
+                                                                                                        WorkflowParser.LATEST_VERSION);
+                    String objectKind = isCatalogObjectExist ? catalogObjectService.getCatalogObjectMetadata(bucketName,
```
Comment (mcompastie) [inline src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java]: Please consider improving code readability by leaving aside tertiary operator.

File: src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java [hunk 2]
```diff
@@ -299,20 +320,33 @@ private void callGraphStyle(JGraphXAdapter<GraphNode, DefaultEdge> callGraphAdap
...
+                                                                         catalogObjectMetadata.getName(),
+                                                                         catalogObjectMetadata.getKind(),
+                                                                         true);
 
                 for (String dependsOnCatalogObject : dependsOnCatalogObjects) {
                     String bucketName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(0);
-                    String workflowName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(1);
-                    GraphNode calledWorkflow = callGraphHolder.addNode(bucketName, workflowName);
-                    callGraphHolder.addDependsOnEdge(callingWorkflow, calledWorkflow);
+                    String objectName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(1);
```
Comment (mcompastie) [inline src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java]: Are you verifying separatorUtility.getSplitBySeparator(dependsOnCatalogObject).length() > 1 ?

File: src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java [hunk 3]
```diff
@@ -185,10 +191,11 @@ private void drawPartitionImageOnSinglePage(BufferedImage partition, PDDocument
         PDRectangle mediaBox = page.getMediaBox();
         float fX = pdImage.getWidth() / mediaBox.getWidth();
         float fY = pdImage.getHeight() / mediaBox.getHeight();
-        float startY = mediaBox.getHeight() - pdImage.getHeight();
+        float startY = pageIndex != 0 ? mediaBox.getHeight() - pdImage.getHeight()
+                                      : mediaBox.getHeight() - (HEADER_HEIGHT + pdImage.getHeight());
         float factor = Math.max(fX, fY);
         if (pageIndex == 0) {
-            factor = factor * 1.15f;
+            factor = factor * 1.10f;
```
Comment (mcompastie) [inline src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java]: You should export this magic number as static & final attribute of this class.

File: src/main/java/org/ow2/proactive/catalog/callgraph/GraphNode.java [hunk 4]
```diff
@@ -92,10 +112,26 @@ public String getObjectName() {
...
     @Override
     public String toString() {
         final StringBuilder sb = new StringBuilder();
-        sb.append(bucketName).append('/').append(objectName);
+        sb.append(" ")
+          .append(bucketName)
+          .append('/')
+          .append(objectName)
+          .append(" ")
+          .append(System.getProperty("line.separator"))
```
Comment (mcompastie) [inline src/main/java/org/ow2/proactive/catalog/callgraph/GraphNode.java]: Please consider System.lineSeparator() .

File: src/main/java/org/ow2/proactive/catalog/callgraph/GraphNode.java [hunk 5]
```diff
@@ -92,10 +112,26 @@ public String getObjectName() {
...
+          .append("[")
+          .append(objectKind)
+          .append("]");
```
Comment (mboussaa) [inline src/main/java/org/ow2/proactive/catalog/callgraph/GraphNode.java]: do you really need many appends? why not one?

File: src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java (hunk 2 above)
Comment (aminelouati) [reply src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java]: dependsOnCatalogObject respect the following model: bucketName/ObjectName

File: src/main/java/org/ow2/proactive/catalog/callgraph/GraphNode.java (hunk 5 above)
Comment (aminelouati) [reply src/main/java/org/ow2/proactive/catalog/callgraph/GraphNode.java]: A single append is possible but for more clarity about the String structure, I used several ones.

**Pull request merged**